### PR TITLE
Enable Batchnorm tests on Navi3x 4x

### PR DESCRIPTION
**Pull request by xinlipn**
## Background:
BatchNorm tests are skipped by a check for CK supported versions.  Current BN code is based on OpenCL and should run on Navi3x and 4x. This fix is to enabled BN tests on these ASICs.

## Proposed changes

The fix is to add Navi3x and 4x to this whitelist is_ck_whitelist(...). Here's related JIRA ticket: LWPMIOPEN-1673

### CODE REVIEW

**Comment by xinlipn** [comment]
Batchnorm tests not skipping on Navi3x

root@aus-navi3x-01:/MIOpen/build# ./bin/miopen_gtest --gtest_filter=*GPU_bn*:*GPU_BN* | tee bntest.log
Running main() from /opt/rocm/cget/build/tmp-c4f1a42bcee536579a1248d25169f834201170c187fc6d80e8c510a7d28f6f48/googletest-1.14.0/googletest/src/gtest_main.cc
PRNG seed: 12345678
Note: Google Test filter = *GPU_bn*:*GPU_BN*
[==========] Running 3130 tests from 42 test suites.
[----------] Global test environment set-up.
[----------] 152 tests from Smoke/GPU_BNOCLBWDLarge2D_FP16
[ RUN      ] Smoke/GPU_BNOCLBWDLarge2D_FP16.BnV2LargeBWDOCL2D_fp16/NCHW_BNSpatial_testBNAPIV2_Dim_2_test_id_0
MIOpen(HIP): Info [get_device_name] Raw device name: gfx1100
MIOpen(HIP): Info [Handle] stream: 0x15fb12e0, device_id: 0
MIOpen(HIP): Info [get_device_name] Raw device name: gfx1100
MIOpen(HIP): Info [Handle] stream: 0x15fb12e0, device_id: 0
MIOpen(HIP): Info [AmdRocmMetadataVersionDetect] ROCm MD version AMDHSA_COv3, HIP version 6.4.43483, MIOpen version 3.4.1.356ce3a20
MIOpen(HIP): Info [FindSolutionImpl] BnBwdTrainingSpatial
MIOpen(HIP): Info [Measure] ReadonlyRamDb::Prefetch time: 7e-05 ms
MIOpen(HIP): Info [Prefetch] File is unreadable: "/MIOpen/build/share/miopen/db/batchnorm_gfx1100_48.HIP.3_4_1.udb.txt"
MIOpen(HIP): Info [Measure] RamDb::Prefetch time: 0.02209 ms
MIOpen(HIP): Info [FindSolutionImpl] Perf Db: record not found for: BnBwdTrainingSpatial
MIOpen(HIP): Info [GetDefaultPerformanceConfig] Variant2-4-1-1024-1-64
MIOpen(HIP): Info [PrintVersionImpl] COMgr v.3.0.0


[       OK ] Smoke/GPU_BNOCLBWDLarge2D_FP16.BnV2LargeBWDOCL2D_fp16/NCHW_BNSpatial_testBNAPIV2_Dim_2_test_id_0 (89105 ms)

**Comment by xinlipn** [comment]
3130 tests passed on Navi3x
[bntest_navi3x.log](https://github.com/user-attachments/files/21308691/bntest_navi3x.log)

**Review by TristanZerrAMD — APPROVED** [review]
LGTM

File: src/include/miopen/solver/ck_utility_common.hpp [hunk 1]
```diff
@@ -42,7 +42,8 @@ namespace ck_utility {
 static inline bool is_ck_whitelist(const std::string& device_name)
 {
     return (StartsWith(device_name, "gfx908") || StartsWith(device_name, "gfx90a") ||
-            StartsWith(device_name, "gfx942") || StartsWith(device_name, "gfx950"));
+            StartsWith(device_name, "gfx942") || StartsWith(device_name, "gfx950") ||
+            StartsWith(device_name, "gfx110") || StartsWith(device_name, "gfx120"));
 }
```
Comment (BrianHarrisonAMD) [inline src/include/miopen/solver/ck_utility_common.hpp]: I believe this will impact solver selection for many CK based solvers.
I think we instead don't want to be using the ck_whitelist check inside the bn tests since they are unrelated to CK now. 

I think the issue is that when we added CK Batchnorm support we changed several things to use this check, and the OCL Batchnorm implementation shouldn't rely on this. Later we fully removed the CK Batchnorm support, and now I think there is nothing in Batch norm that should be checking CK's applicability. 

Somewhat surprised this isn't breaking other tests, but the CK solvers typically also check the instances returned by CK, and will also say not applicable if there isn't any.

**Comment by xinlipn** [comment]
Removed is_ck_whitelist(...) for BN tests. Tested on gfx1100/Navi31, BN test is not skipped

root@aus-navi3x-01:/MIOpen/build# ./bin/test_bn_bwd
Running main() from /opt/rocm/cget/build/tmp-c4f1a42bcee536579a1248d25169f834201170c187fc6d80e8c510a7d28f6f48/googletest-1.14.0/googletest/src/gtest_main.cc
[==========] Running 512 tests from 7 test suites.
[----------] Global test environment set-up.
[----------] 152 tests from Smoke/GPU_BNOCLBWDLarge2D_FP16
[ RUN      ] Smoke/GPU_BNOCLBWDLarge2D_FP16.BnV2LargeBWDOCL2D_fp16/NCHW_BNSpatial_testBNAPIV2_Dim_2_test_id_0
PRNG seed: 12345678
MIOpen(HIP): Info [get_device_name] Raw device name: gfx1100
MIOpen(HIP): Info [Handle] stream: 0x34f8b500, device_id: 0
MIOpen(HIP): Info [AmdRocmMetadataVersionDetect] ROCm MD version AMDHSA_COv3, HIP version 6.4.43483, MIOpen version 3.4.1.e8fb142e5-dirty
MIOpen(HIP): Info [FindSolutionImpl] BnBwdTrainingSpatial
MIOpen(HIP): Info [Measure] ReadonlyRamDb::Prefetch time: 0.00014 ms
MIOpen(HIP): Info [Prefetch] File is unreadable: "/MIOpen/build/share/miopen/db/batchnorm_gfx1100_48.HIP.3_4_1.udb.txt"
MIOpen(HIP): Info [Measure] RamDb::Prefetch time: 0.03073 ms
MIOpen(HIP): Info [FindSolutionImpl] Perf Db: record not found for: BnBwdTrainingSpatial
MIOpen(HIP): Info [GetDefaultPerformanceConfig] Variant2-4-1-1024-1-64
MIOpen(HIP): Info [PrintVersionImpl] COMgr v.3.0.0

[       OK ] Smoke/GPU_BNOCLBWDLarge2D_FP16.BnV2LargeBWDOCL2D_fp16/NCHW_BNSpatial_testBNAPIV2_Dim_2_test_id_0 (88333 ms)

**Review by BrianHarrisonAMD — APPROVED** [review]
